Proposed: Add a doc about `.editorconfig`

ggwicz opened the pull request:
I noticed that the `.editorconfig` in the [solidwp-website](https://github.com/stellarwp/solidwp-website) project is the same as the one in [orderable-website](https://github.com/stellarwp/orderable-website), and figured it could be worth defining this somewhere for consistency.

We're using similar or outright-the-same rules elsewhere and it felt worth establishing a "standard".

At the same time, it doesn't feel like a "big" enough deal to warrant e.g. its own repo or something, and given how simple the file is, an `.md` file with the contents in a code block **seemed** like a sufficient-enough reference to me...

What do you think?

borkweb commented on docs/editorconfig.md:
```diff
@@ -0,0 +1,38 @@
+# https://github.com/stellarwp/global-docs/blob/main/docs/editorconfig.md
+# https://editorconfig.org
+
+root = true
+
+[*]
+charset = utf-8
+end_of_line = lf
+indent_size = 4
+tab_width = 4
+indent_style = space
+insert_final_newline = true
+trim_trailing_whitespace = true
+
+[*.txt]
+trim_trailing_whitespace = false
```
Here's a more common default!

```suggestion
root = true

[*]
charset = utf-8
end_of_line = lf
indent_style = tab
insert_final_newline = true
tab_width = 4
trim_trailing_whitespace = true

[**.{jshintrc,json,scss-lint,yml}]
indent_style = space
indent_size = 4

[*.txt]
trim_trailing_whitespace = false
```

borkweb reviewed (changes requested):
I like it, though the defaults should switch to our more common setup (WP-based)

ggwicz commented:
Awesome @borkweb, thank you!

I've implemented the updated `.editorconfig`, changed the doc title from `Editorconfig` to the more-correct `EditorConfig`, and simplified the language in the doc summary a bit.

Mind giving this all another review?
